lib/Pinto/Action/Pull.pm

```diff
@@ -45,7 +45,7 @@ with qw( Pinto::Role::Committable Pinto::Role::Puller );
 sub BUILD {
     my ($self) = @_;
 
-    $self->stack->assert_not_locked;
+    $self->stack->assert_not_locked unless $self->dry_run;
```

Comment:
right, was narrowly considering "add" - does seem more contained and obvious in Committable, i'll give it a go